org.eclipse.jdt.core/dom/org/eclipse/jdt/internal/core/dom/rewrite/ASTRewriteAnalyzer.java

```diff
@@ -603,6 +603,10 @@ protected boolean mustRemoveSeparator(int originalOffset, int nodeIndex) {
 			return true;
 		}
 
+		protected boolean mustRemoveEndKeywordWhenEmpty() {
```

Comment:
According the issue description, the last part we want to remove could be a token as well right ? Should we call this `LastToken` ? Or did you misunderstood the issue and naming used here :(

Reply:
Hi @gayanper I was basing on the passing of keyword/end-keyword to the rewriteList() method from try-with-resources where ")" was the end-keyword for the resources plus the fact that the remove all was commented as removing the keyword.  If you feel that should be mustRemoveLastToken() I am fine with renaming.  I could also change the arguments to pass in a token to remove so that it ensures that token is removed plus any preceding comments.  I assumed that an invalid token at the end of the resources list couldn't occur such that a removal of resources would be possible, but again I may be assuming to much.

Reply:
I think the current name looks good according to the above comment.